sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -222,10 +245,19 @@ case class AlterViewAsCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
-    if (session.sessionState.catalog.isTemporaryTable(name)) {
-      session.sessionState.catalog.createTempView(name.table, analyzedPlan, overrideIfExists = true)
-    } else {
+    if (name.database.isDefined) {
       alterPermanentView(session, analyzedPlan)
+    } else if (session.sessionState.catalog.isTemporaryTable(name)) {
```

Comment:
Shouldn't we check the current database for the view name ? Or do we assume that database is resolved into the tableidentifier ?